oqs/wrapper.py

```diff
@@ -321,3 +309,38 @@ def free(self):
 
     def __repr__(self):
         return "Signature mechanism: " + self._sig.contents.method_name.decode()
+
+
+liboqs.OQS_SIG_new.restype = ct.POINTER(OQS_SIG)
+liboqs.OQS_SIG_alg_identifier.restype = ct.c_char_p
+
+
+def is_sig_enabled(alg_name):
+    """Returns True if the signature algorithm is enabled, False otherwise.
+
+    Attribute:
+        alg_name -- a signature mechanism algorithm name
+    """
+    try:
+        sig = liboqs.OQS_SIG_new(ct.create_string_buffer(alg_name.encode()))
+        if(sig.contents):
+            liboqs.OQS_SIG_free(sig)
+            return True
+    except ValueError:
+        pass
+    return False
+
+
+_sig_alg_ids = [liboqs.OQS_SIG_alg_identifier(i) for i in range(liboqs.OQS_SIG_alg_count())]
+_supported_sigs = [i.decode() for i in _sig_alg_ids]
+_enabled_sigs = [i for i in _supported_sigs if is_sig_enabled(i)]
+
+
+def get_enabled_sig_mechanisms():
+    """Returns the list of enabled signature mechanisms."""
+    return _enabled_sigs
+
+
+def print_enabled_sig_mechanisms():
```

Comment:
Do we need to have print_enabled_sig_mechanisms? Doesn't seem like a "fundamental" method in the class.

Reply:
Same comment as KEM

Reply:
Fixed too.